tests/TokenConverter/AbstractTokenConverter.ts

```diff
@@ -0,0 +1,693 @@
+
+      await expect(
+        converter.convertExactTokens(
+          convertToUnit(".25", 18),
+          convertToUnit(".5", 18),
+          tokenInDeflationary.address,
+          tokenOut.address,
+          await to.getAddress(),
+        ),
+      ).to.be.revertedWithCustomError(converter, "AmountInMismatched");
+    });
+  });
+
+  describe("Convert tokens for exact tokens with supporting fee", async () => {
+    beforeEach(async () => {
+      await destination.convertibleBaseAsset.returns(tokenIn.address);
+      await tokenInDeflationary.connect(owner).approve(converter.address, convertToUnit("1", 18));
+      await tokenIn.connect(owner).approve(converter.address, convertToUnit("1", 18));
+      await tokenOut.transfer(converter.address, convertToUnit("1.5", 18));
+
+      await oracle.getPrice.whenCalledWith(tokenInDeflationary.address).returns(TOKEN_IN_PRICE);
+      await oracle.getPrice.whenCalledWith(tokenIn.address).returns(TOKEN_IN_PRICE);
+      await oracle.getPrice.whenCalledWith(tokenOut.address).returns(TOKEN_OUT_PRICE);
+    });
+
+    it("Revert on lower amount out than expected", async () => {
+      await converter.setConversionConfig(tokenIn.address, tokenOut.address, ConversionConfig);
+      await expect(
+        converter.convertForExactTokensSupportingFeeOnTransferTokens(
+          convertToUnit(".5", 18),
+          convertToUnit("1.5", 18),
+          tokenIn.address,
+          tokenOut.address,
+          await to.getAddress(),
+        ),
+      ).to.be.revertedWithCustomError(converter, "AmountInHigherThanMax");
+    });
+
+    it("Revert when address(to) is same as tokenAddressIn or tokenAddressOut", async () => {
+      await converter.setConversionConfig(tokenIn.address, tokenOut.address, ConversionConfig);
+
+      await expect(
+        converter.convertForExactTokensSupportingFeeOnTransferTokens(
+          convertToUnit(".5", 18),
+          convertToUnit("1.5", 18),
+          tokenIn.address,
+          tokenOut.address,
+          tokenIn.address,
+        ),
+      ).to.be.revertedWithCustomError(converter, "InvalidToAddress");
+    });
+
+    it("Revert on deflationary token transfer", async () => {
+      await destination.convertibleBaseAsset.returns(tokenInDeflationary.address);
+
+      const ConversionConfig = {
+        incentive: INCENTIVE,
+        enabled: true,
+      };
+
+      await converter.setConversionConfig(tokenInDeflationary.address, tokenOut.address, ConversionConfig);
+
+      await expect(
+        converter.convertForExactTokens(
+          convertToUnit(".25", 18),
+          convertToUnit(".5", 18),
+          tokenInDeflationary.address,
+          tokenOut.address,
+          await to.getAddress(),
+        ),
+      ).to.be.revertedWithCustomError(converter, "AmountOutMismatched");
+    });
+
+    // We can not keep convertForExactTokens with supporting fee.
```

Comment:
`convertForExactTokensSupportingFeeOnTransferTokens` will not work for deflationary tokens as it takes the exact amount the user wants but if `tokenIn` is deflationary then the amount received by converter will be less than expected. Due to this it will also send less amount to user not the amount expected by user.
So, i commented these test cases for now. I will remove them.